0.29.0 (#1187)
MetaMask/snaps-monorepo: packages/rpc-methods/CHANGELOG.md

## [Unreleased]

## [0.29.0]
### Added
- **BREAKING:** Improve request function types ([#1014](https://github.com/MetaMask/snaps-monorepo/pull/1014))

>   - `snap.request` is now strongly typed in line with the rpc methods in this package

Reviewer: ```suggestion
  - `snap.request` is now strongly typed in line with the JSON-RPC methods in this package
```

## [Unreleased]

## [0.29.0]
### Added

> - **BREAKING:** `snap.request` is now strongly typed in line with the rpc methods in this package ([#1014](https://github.com/MetaMask/snaps-monorepo/pull/1014))

Reviewer: ```suggestion
- **BREAKING:** Improve request function types ([#1014](https://github.com/MetaMask/snaps-monorepo/pull/1014))
  - `snap.request` is now strongly typed in line with the JSON-RPC methods in this package
```

### Changed

> - **BREAKING:** `wallet_snap_*` rpc method was deprecated in favor of `wallet_snap` and new caveat type, `snapIds`, was added ([#1182](https://github.com/MetaMask/snaps-monorepo/pull/1182))

Reviewer: I don't think `wallet_snap_*` still works, so it's not really deprecated, is it?

```suggestion
- **BREAKING:** `Replace `wallet_snap_*` permission with `wallet_snap` ([#1182](https://github.com/MetaMask/snaps-monorepo/pull/1182))
  - The new permission uses a caveat-based implementation to specify the snaps that are allowed to be interacted with
```

> - Standardise logging across all packages ([#1164](https://github.com/MetaMask/snaps-monorepo/pull/1164))
> - Bump @metamask/permission-controller from 1.0.2 to 2.0.0 ([#1156](https://github.com/MetaMask/snaps-monorepo/pull/1156))
> - Bump `@metamask/key-tree` to 6.2.1 ([#1142](https://github.com/MetaMask/snaps-monorepo/pull/1142))

Reviewer: These are internal changes, and don't need to be mentioned in the changelog.

```suggestion
```

> ### Fixed
> - Fix typo in documentation ([#1140](https://github.com/MetaMask/snaps-monorepo/pull/1140))
> 

Reviewer: ```suggestion
```
